### PR TITLE
(GH-530) Enable Extension on Workspace files

### PR DESCRIPTION
**Pull request by jpogran**
This PR adds a file watcher to the `activationEvents` that initializes the extension if there are any Puppet or metadata.json files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,6 +53,8 @@
   "activationEvents": [
     "onLanguage:puppet",
     "onLanguage:puppetfile",
+    "workspaceContains:**/*.pp",
+    "workspaceContains:**/metadata.json",
```
Comment (glennsarti) [inline package.json]: Curious as to the reason for the permissive glob instead of just `metadata.json`

File: package.json [hunk 2]
```diff
@@ -53,6 +53,8 @@
   "activationEvents": [
     "onLanguage:puppet",
     "onLanguage:puppetfile",
+    "workspaceContains:**/*.pp",
```
Comment (glennsarti) [inline package.json]: Add epp as well?

Comment (glennsarti) [reply package.json]: Puppetfile ?

File: package.json (hunk 1 above)
Comment (jpogran) [reply package.json]: Was thinking to catch situations like control repo, where the metadata or manifest is several levels deep

File: package.json (hunk 2 above)
Comment (jpogran) [reply package.json]: Added

**Comment by jpogran** [comment]
The current question with this PR is to leave `metadata.json` in it or not.

If `metadata.json` is not in this PR, any folder opened without Puppet manifest files won't load without opening a Puppet manifest file on purpose. Modules like the puppetlabs-powershell module do not have Puppet manifests, but do have `metadata.json` files.  If `metadata.json` is left in this PR, then any module will activate the extension.

There is risk that there is something else out there that uses a file called `metadata.json` and would cause our extension to activate when they did not intend to use it

**Comment by glennsarti** [comment]
Other examples of metadata.json (Google search)

https://www.vagrantup.com/docs/boxes/format.html
https://www.chromium.org/chromium-os/build/cbuildbot-metadata-json-tutorial
https://2isa.readthedocs.io/en/latest/other/json-metadata.html
https://gitlab.gnome.org/World/ShellExtensions/desktop-icons/blob/master/metadata.json
https://fabric-shim.github.io/release-1.4/tutorial-annotated-contract-metadata.html

TBH, I'm leaning towards leave it out of this PR and raise an issue to spike trying to figure out a better way.

**Comment by jpogran** [comment]
@glennsarti agreed. Amended commit and ready for merge

**Pull request merged**